store/src/java/com/zimbra/qa/unittest/TestImapNotificationsViaEmbeddedLocal.java

```diff
@@ -0,0 +1,47 @@
+package com.zimbra.qa.unittest;
+
+import static org.junit.Assert.assertFalse;
+
+import java.io.IOException;
+import java.util.Set;
+
+import org.dom4j.DocumentException;
+import org.junit.After;
+import org.junit.Before;
+
+import com.zimbra.client.ZFolder;
+import com.zimbra.client.ZMailbox;
+import com.zimbra.common.localconfig.ConfigException;
+import com.zimbra.common.localconfig.LC;
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.cs.imap.ImapRemoteSession;
+import com.zimbra.cs.imap.ImapServerListener;
+import com.zimbra.cs.imap.ImapServerListenerPool;
+
+public class TestImapNotificationsViaEmbeddedLocal extends SharedImapNotificationTests {
+
+    @Before
+    public void setUp() throws ServiceException, IOException, DocumentException, ConfigException  {
+        saveImapConfigSettings();
+        TestUtil.setLCValue(LC.imap_always_use_remote_store, String.valueOf(false));
+        imapServer.setReverseProxyUpstreamImapServers(new String[] {});
+        super.sharedSetUp();
```

Comment:
This test should use assumptions to check that embedded local IMAP is enabled. If you install remote IMAP and disable local one, all tests in this suite will fail.